runtime/runtime-params-estimator/estimator-warehouse/src/check.rs

```diff
@@ -131,6 +150,52 @@ fn estimation_changes(
     Ok(warnings)
 }
 
+fn estimation_uncertain_changes(
+    db: &Db,
+    estimation_names: &[String],
+    commit_before: &str,
+    commit_after: &str,
+    metric: Metric,
+) -> anyhow::Result<Vec<Notice>> {
+    let mut warnings = Vec::new();
+    for name in estimation_names {
+        let b = &EstimationRow::get(db, name, commit_before, metric)?[0];
+        let a = &EstimationRow::get(db, name, commit_after, metric)?[0];
+        match (&b.uncertain_reason, &a.uncertain_reason) {
+            (None, None) => continue,
+            (Some(uncertain_before), None) => add_warning(
+                &mut warnings,
+                name.clone(),
+                uncertain_before.clone(),
+                "None".to_owned(),
+            ),
+            (None, Some(uncertain_after)) => {
+                add_warning(&mut warnings, name.clone(), "None".to_owned(), uncertain_after.clone())
+            }
+            (Some(uncertain_before), Some(uncertain_after)) => {
+                if !uncertain_before.eq(uncertain_after) {
+                    add_warning(
+                        &mut warnings,
+                        name.clone(),
+                        uncertain_before.clone(),
+                        uncertain_after.clone(),
+                    );
+                }
+            }
+        }
+    }
```

Comment:
The uncertain reason strings are cloned unnecessarily here. It's not actually performance critical-code, so it doesn't matter too much. But for the sake of clean code, I would still use something like this
```rust
        let b = EstimationRow::get(db, name, commit_before, metric)?.remove(0);
        let a = EstimationRow::get(db, name, commit_after, metric)?.remove(0);
        match (b.uncertain_reason, a.uncertain_reason) {  
```
... and remove the clones afterwards.

Reply:
Fixed